atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraClientRateLimitingConfig.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.cassandra;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.immutables.value.Value;
+
+@JsonDeserialize(as = ImmutableCassandraClientRateLimitingConfig.class)
+@JsonSerialize(as = ImmutableCassandraClientRateLimitingConfig.class)
+@Value.Immutable
+public interface CassandraClientRateLimitingConfig {
+    @Value.Default
+    default int maxConcurrentRangeScans() {
+        return 1_000;
```

Comment:
Do we have a sense of how many concurrent range scans a client might be firing?

Reply:
this PR adds metrics which should give us an actual read - we do have some existing internal metrics which show the rate of range scans, but that's slightly different from concurrency at a point in time

I could bump this to be much higher to start with, which means this will have no immediate effect but give us the metrics

Reply:
yes, I'd like us to be more cautious here to start with, we can be more aggressive as soon as we have metrics.

Reply:
bumped up a lot